test/mocks/malicious/returnBomb.t.sol

```diff
@@ -0,0 +1,18 @@
+pragma solidity ^0.8.0;
+
+import "forge-std/Test.sol";
+
+import {returnBomb} from "../../../src/mocks/malicious/returnBomb.sol";
+
+contract returnBombTest is Test {
+    returnBomb public attacker;
+
+    function setUp() public {
+        attacker = new returnBomb();
+    }
+
+    function testReturnBomb() public {
+        (bool success, bytes memory returnData) = address(attacker).call{gas: 3397}("");
```

Comment:
We should be check that a function that has the low level call reverts, since the assumption is when a low level call is used reverts will not bubble up.
e.g.

```
function testReturnBomb() public {
  vm.expectRevert();
  callWithGasLimit(3397);
}

function callWithGasLimit(uint256 _gas) public {
  // This function should never revert since a low level call is used
  (bool success, bytes memory returnData) = address(attacker).call{gas: _gas}("");
}
```

Reply:
There's a sweet spot of revert data where if the returned data is too small, we won't cause an OOG error in the outer call, but if the returned data is too large, we run out of gas in the inner call and end up returning no data to the outer call and don't cause it to revert.